Detectors/Upgrades/ALICE3/Passive/src/Pipe.cxx

```diff
@@ -93,75 +94,140 @@ void Alice3Pipe::ConstructGeometry()
-  // Strategy used here is to use a composite shape where shapes of TRK layers are subtracted to the vacuum volume
-  TGeoTube* outerBeTube = new TGeoTube("OUT_PIPEsh", mBeOuterPipeRmin, mBeOuterPipeRmin + mBeOuterPipeThick, mOuterIpHLength);
-  TGeoVolume* outerBeTubeVolume = new TGeoVolume("OUT_PIPE", outerBeTube, kMedBe);
-  outerBeTubeVolume->SetLineColor(kGreen - 9);
-
-  TGeoTube* outerBerylliumTubeVacuumBase = new TGeoTube("OUT_PIPEVACUUM_BASEsh", mBeInnerPipeRmin + mBeInnerPipeThick, mBeOuterPipeRmin, mOuterIpHLength); // Vacuum filling for outer pipe
-  TGeoCompositeShape* outerBerylliumTubeVacuumComposite;                                                                                                   // Composite volume to subctract to vacuum
-  TGeoVolume* outerBerylliumTubeVacuumVolume;                                                                                                              // Final volume to be used
+  // We split the naming of the parts if the beam pipe for ALICE 3 into parts
+  // - pipe
+  // - vacuum vessel (which hosts the primary vacuum)
+  // - iris vacuum vessel (which hosts the secondary vacuum)
 
-  TString compositeFormula{"OUT_PIPEVACUUM_BASEsh"}; // If pipe is alone we won't subctract anything
+  // A3IP update
+  // Vacuum
+  TGeoTube* vacuumBasePipe = new TGeoTube("PIPEVACUUM_BASEsh", 0., mPipeRIn, mA3IPLength / 2.);
+  TGeoTube* vacuumBaseVacuumVessel = new TGeoTube("VACUUM_VESSELVACUUM_BASEsh", mPipeRIn, mVacuumVesselRIn, mVacuumVesselLength / 2.);
+  // Excavate volumes from the vacuum such that there is place for the TRK barrel layers and FT3 disc layers of the IRIS tracker
+  // And the other passive shapes: coldplate, iris tracker vacuum vessel
+  TGeoCompositeShape* vacuumComposite;
+  TGeoVolume* vacuumVolume;
+  TString compositeFormula{"PIPEVACUUM_BASEsh+VACUUM_VESSELVACUUM_BASEsh"};
   TString subtractorsFormula;
 
   if (!mIsTRKActivated) {
     std::vector<TGeoTube*> trkLayerShapes;
 
-    std::vector<std::array<float, 3>> layersQuotas = {std::array<float, 3>{0.5f, 50.f, 50.e-4}, // TODO: Set layers dynamically. {radius, zLen, thickness}
-                                                      std::array<float, 3>{1.2f, 50.f, 50.e-4},
-                                                      std::array<float, 3>{2.5f, 50.f, 50.e-4}};
+    std::vector<std::array<float, 3>> layersQuotas = {std::array<float, 3>{0.5f, 50.f, 100.e-4}, // TODO: Set layers dynamically. {radius, zLen, thickness}
+                                                      std::array<float, 3>{1.2f, 50.f, 100.e-4},
+                                                      std::array<float, 3>{2.5f, 50.f, 100.e-4}};
 
-    subtractorsFormula = "TRKLAYER_0sh";                           // First volume to be subctracted (at least one has to be provided)
+    subtractorsFormula = "";                                       // First volume to be subctracted (at least one has to be provided)
```

Comment:
I don't see what you gain from this rewritten logic, but perhaps this line should then go, no?
```suggestion
```

Reply:
For the FT3 disc layers L0 - L2, space needs to be excavated from the vacuum as well. If you were to run the simulation with only A3IP and FT3, but not the TRK, then it does not do that and you create an overlap.
Now it can be run any way

Reply:
Yes, the comment should go

Reply:
I wanted the code to be consistent with the part where the volumes for L0 - L2 of the FT3 get excavated from the vacuum.

In addition, since you can specify the build of the TRK from a cfg file, I thought, if no layers are provided, it would only produce the iris/inner vacuum vessel